TEAMMATES/teammates
# [#9536] Instructor Feedback Sessions Page E2E Test

Part of #9536

**Outline of Solution**

- Add methods to `BaseE2ETestCase` to verify downloaded files exist
- Add methods to `BaseE2ETestCase` to compare emails sent
- Add `InstructorFeedbackSessionsPage` to represent instructor sessions page
- Add relevant backdoor methods to compare feedback session data in datastore
- Add `InstructorFeedbackSessionsPageE2ETest` to test:
    - verify loaded data
    - verify response rate
    - add new session
    - add new copied session
    - copy session
    - publish results
    - send reminder email
    - resend results link
    - unpublish results
    - download results
    - soft delete session
    - restore session
    - permanently delete session
    - restore all session
    - delete all session
